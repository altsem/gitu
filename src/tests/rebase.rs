use super::*;

fn setup_rebase() -> (TestContext, crate::state::State) {
    let mut ctx = TestContext::setup_clone(80, 20);
    run(ctx.dir.path(), &["git", "checkout", "-b", "other-branch"]);
    run(ctx.dir.path(), &["git", "checkout", "main"]);
    commit(ctx.dir.path(), "new-file", "hello");
    run(ctx.dir.path(), &["git", "checkout", "other-branch"]);

    let state = ctx.init_state();
    (ctx, state)
}

#[test]
fn rebase_menu() {
    let (mut ctx, mut state) = setup_rebase();

    state.update(&mut ctx.term, &keys("r")).unwrap();
    insta::assert_snapshot!(ctx.redact_buffer());
}

#[test]
fn rebase_elsewhere_prompt() {
    let (mut ctx, mut state) = setup_rebase();

    state.update(&mut ctx.term, &keys("re")).unwrap();
    insta::assert_snapshot!(ctx.redact_buffer());
}

#[test]
fn rebase_elsewhere() {
    let (mut ctx, mut state) = setup_rebase();

    state.update(&mut ctx.term, &keys("remain<enter>")).unwrap();
    insta::assert_snapshot!(ctx.redact_buffer());
}
