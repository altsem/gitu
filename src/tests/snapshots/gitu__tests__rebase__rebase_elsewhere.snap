---
source: src/tests/rebase.rs
expression: ctx.redact_buffer()
---
▌On branch other-branch                                                         |
                                                                                |
 Recent commits                                                                 |
 46c81ca main other-branch add new-file                                         |
 b66a0bf origin/main add initial-file                                           |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
────────────────────────────────────────────────────────────────────────────────|
$ git rebase --autostash main                                                   |
Successfully rebased and updated refs/heads/other-branch.                       |
styles_hash: b7a28b018f24d4c7
